packages/push/src/PushRegistration.ts

```diff
@@ -96,22 +97,102 @@ export class PushRegistration {
       "categories": categories
     };
 
-    const instance = axios.create({
+    const http = axios.create({
       baseURL: url,
       timeout: 5000,
       headers: {"Authorization": `Basic ${authToken}`}
     });
 
     return new Promise((resolve, reject) => {
-      return instance.post(PushRegistration.API_PATH, postData)
+      return http.post(PushRegistration.API_PATH, postData)
       .then(
         () => {
+          const storage = window.localStorage;
+          storage.setItem(PushRegistration.TOKEN_KEY, deviceToken);
+
           if (isCordovaAndroid()) {
             this.subscribeToFirebaseTopic(variantId);
             for (const category of categories) {
               this.subscribeToFirebaseTopic(category);
             }
           }
+
+          resolve();
+        }
+      )
+      .catch(reject);
+    });
+  }
+
+  /**
+   * Register deviceToken for Android or IOS platforms
+   */
+  public unregister(): Promise<void> {
```

Comment:
How does this should work? Usually unregister is called when app is uninstalled but how do you plan to hook into the native callback. 

Unregister is usually done on native side.

Reply:
@wtrocki Sorry, not sure if I understood your question here. Do you mean in which case/scenario the unregister should be used?

Reply:
Is there any requirement for that or any use case that you see people to use it?
Do we plan to add that to showcase or explain how unregister should be used?

Reply:
The requirement is that this feature is available on UPS from ages and also in all old SDKs. No, it was not added only because of the showcase switch [register/unregister]

Maybe _application will rarely have the switch like in showcase_ but I can give you a real scenario. After login, register the device after logout unregister the device but maybe the community has a bunch of real examples. That is a good thing to ask on ML.

When an app is deleted the device is not going to receive any messages anymore but the device will still "registered/listed" on UPS. The developer can try to unregister device when the app is deleted but unfortunately, this feature is only available on Android (as far as I know)

UPS will automatically purge the device from the database when Firebase return saying that is an invalid token and we also have a way to [purge an old token when the device registers the app again](https://github.com/aerogear/aerogear-unifiedpush-server/blob/ff5ba2cb8446d929e5fd426bcb645299b4559914/jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java#L132)